core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -693,6 +735,25 @@ public Custom getCustom(String type) {
             return this.customs.get(type);
         }
 
+        public Builder putActiveAllocationIds(int shardId, AllocationIds allocationIds) {
+            activeAllocationIds.put(shardId, allocationIds);
+            return this;
+        }
+
+        public Builder removeActiveAllocationIds(int shardId) {
```

Comment:
these do not seem to be used. I don't thing we need them? I rather make sure there is always a set for each shard.